wondrous-app/components/Common/TaskViewModalAutocomplete/index.tsx

```diff
@@ -0,0 +1,101 @@
+import { Box, Grid } from '@mui/material';
+import { SafeImage } from 'components/Common/Image';
+import DefaultUserImage from 'components/Common/Image/DefaultUserImage';
+import SearchIcon from 'components/Icons/search';
+import { Arrow } from 'components/Icons/sections';
+import { useState } from 'react';
+import palette from 'theme/palette';
+import { ArrowWrapper, Option, PaperComponent, StyledAutocomplete, StyledTextField } from './styles';
+
+const profilePictureStyle = {
+  display: 'flex',
+  width: '24px',
+  height: '24px',
+  borderRadius: '50%',
+};
+
+const ProfilePicture = ({ profilePicture }) => (
+  <SafeImage
+    src={profilePicture}
+    placeholderComp={<DefaultUserImage style={profilePictureStyle} />}
+    width="24px"
+    height="24px"
+    objectFit="cover"
+    useNextImage
+    style={profilePictureStyle}
+  />
+);
+
+const RenderOption = (props, option) => {
+  const { profilePicture, label, hide } = option || {};
+  if (hide) return null;
+  return (
+    <Option {...props}>
+      <ProfilePicture profilePicture={profilePicture} />
+      {label}
+    </Option>
+  );
+};
+
+const assignToSelf = ({ user, onClick }) => {
```

Comment:
can we move the user check inside the component itself and pass user / onclick with props this way avoiding the assignToSelf function and simply having AssignToSelf component? imo more readable and I might be wrong but don't think we need a wrapper function here